### PR TITLE
keep sarama default setting ,use hash key for partition

### PR DESCRIPTION
**Pull request by timchenxiaoyu**

### CODE REVIEW

File: plugins/flusher/kafka/flusher_kafka.go [hunk 1]
```diff
@@ -27,14 +27,15 @@ import (
...
-	Topic        string
-	ClientID     string
-	isTerminal   chan bool
-	producer     sarama.AsyncProducer
+	context         ilogtail.Context
+	Brokers         []string
+	SASLUsername    string
+	SASLPassword    string
+	Topic           string
+	PartitionerType string
```
Comment (shabicheng) [inline plugins/flusher/kafka/flusher_kafka.go:35]: in init function, set PartitionerType default value

```
return &FlusherKafka{
			ClientID: "LogtailPlugin",
                         PartitionerType: "random",
		}
```

File: plugins/flusher/kafka/flusher_kafka.go [hunk 2]
```diff
@@ -55,7 +56,14 @@ func (k *FlusherKafka) Init(context ilogtail.Context) error {
 	config.ClientID = k.ClientID
 	config.Producer.Return.Successes = true
 	// config.Producer.RequiredAcks = sarama.WaitForAll
-	config.Producer.Partitioner = sarama.NewRandomPartitioner
+	partitioner := sarama.NewRandomPartitioner
+	switch k.PartitionerType {
+	case "roundrobin":
```
Comment (shabicheng) [inline plugins/flusher/kafka/flusher_kafka.go:61]:
```
case "random":
    partitioner := sarama.NewRandomPartitioner
default:
     logger.Error(k.context.GetRuntimeContext(), "INVALID_KAFKA_PARTITIONER", "invalid PartitionerType, use RandomPartitioner instead, type", k.PartitionerType)
```

**Comment by shabicheng** [comment]
LGTM

**Comment by EvanLjp** [comment]
1. Cloud you merge the main branch of upstream to trigger CI process ?
2. Please add change log.

```
git remote add upstream git@github.com:alibaba/ilogtail.git
git fetch upstream
git merge upstream/main
git commit -am "some git logs"
git push origin keep_default_hashkey_partiton

```

**Pull request merged**